src/test/java/me/gnat008/perworldinventory/config/SettingsConsistencyTest.java

```diff
@@ -17,37 +25,110 @@
 import static org.hamcrest.Matchers.empty;
 import static org.hamcrest.Matchers.equalTo;
 import static org.junit.Assert.assertThat;
+import static org.junit.Assert.fail;
 
 /**
  * Tests that the config.yml file corresponds with the settings holder classes in the code.
  */
 public class SettingsConsistencyTest {
 
     /** Bukkit's FileConfiguration#getKeys returns all inner nodes also. We want to exclude those in tests. */
-    private static final List<String> YAML_INNER_NODES = ImmutableList.of("player", "player.stats");
+    private static final List<String> YAML_INNER_NODES = ImmutableList.of("player", "player.stats", "database");
+    private static final String SETTINGS_FOLDER = TestHelper.PROJECT_ROOT + "config";
 
-    private final ConfigurationData configData = ConfigurationDataBuilder.collectData(PwiProperties.class);
-    private final FileConfiguration ymlConfiguration = YamlConfiguration.loadConfiguration(getJarFile("/config.yml"));
+    private static ConfigurationData configData;
+    private static YamlFileResource yamlResource;
+    private static List<Class<? extends SettingsHolder>> classes;
 
+    @BeforeClass
+    public static void scanForSettingsClasses() {
+        ClassCollector collector = new ClassCollector(TestHelper.SOURCES_FOLDER, SETTINGS_FOLDER);
+        classes = collector.collectClasses(SettingsHolder.class);
+
+        if (classes.isEmpty()) {
+            throw new IllegalStateException("Did not find any SettingsHolder classes. Is the folder correct?");
+        }
+
+        System.out.println("Found " + classes.size() + " SettingsHolder implementations");
+
+        configData = SettingsRetriever.buildConfigurationData();
+        yamlResource = new YamlFileResource(getJarFile("/config.yml")); //YamlConfiguration.loadConfiguration(getJarFile("/config.yml"));
+    }
+
+    /**
+     * Make sure that all properties in the config contain a default value.
+     */
     @Test
     public void shouldContainAllPropertiesWithSameDefaultValue() {
         // given / when / then
         for (Property<?> property : configData.getProperties()) {
             assertThat("config.yml does not have property for " + property,
-                ymlConfiguration.contains(property.getPath()), equalTo(true));
+                yamlResource.contains(property.getPath()), equalTo(true));
             assertThat("config.yml does not have same default value for " + property,
-                property.getDefaultValue(), equalTo(ymlConfiguration.get(property.getPath())));
+                property.getDefaultValue(), equalTo(property.getValue(yamlResource)));
         }
     }
 
+    /**
+     * Make sure that all properties defined in both the config and the code.
```

Comment:
This especially checks that config.yml does not have any properties that are unknown in the code. The above method checks the other way around: that config.yml contains all properties as defined in the code.